scripts/add_gen3_repo.py

```diff
@@ -0,0 +1,149 @@
+        # Our refcats and defects are temporary files, and must not be linked
+        _migrate_gen2_to_gen3(dataset, gen2_repo, gen2_calibs, gen3_repo, mode="copy",
+                              config_file="convertRepo_copied.py")
+
+    log.info("Exporting Gen 3 registry to configure new repos...")
+    _export_for_copy(dataset, gen3_repo)
+
+
+def _migrate_gen2_to_gen3(dataset, gen2_repo, gen2_calib_repo, gen3_repo, mode, config_file):
+    """Convert a Gen 2 repository into a Gen 3 repository.
+
+    Parameters
+    ----------
+    dataset : `lsst.ap.verify.dataset.Dataset`
+        The dataset being migrated.
+    gen2_repo, gen2_calib_repo : `str`
+       The locations of the original repositories.
+    gen3_repo : `str`
+       The location of the Gen 3 repository. Must exist, but need not be
+       initialized as a repository.
+    mode : {'relsymlink', 'copy'}
+       Whether the Gen 3 repo should contain symbolic links to the Gen 2
+       datasets, or an independent copy.
+    mode : `str`
+       The config file (in the dataset config directory) with a configuration
+       for `~lsst.obs.base.gen2to3.ConvertRepoTask`
+    """
+    instrument = _get_instrument_class(dataset.camera)
+    config = os.path.join(dataset.configLocation, config_file)
+
+    # calib/<instrument>, refcats, and skymaps collections created by default
+    convertGen2RepoToGen3.convert(gen2_repo, gen3_repo, instrument,
+                                  # skymap detected automatically; do nothing
+                                  calibs=gen2_calib_repo,
+                                  config=config,
+                                  transferMode=mode)
+
+
+def _export_for_copy(dataset, repo):
+    """Export a Gen 3 repository so that a dataset can make copies later.
+
+    Parameters
+    ----------
+    dataset : `lsst.ap.verify.dataset.Dataset`
+        The dataset needing the ability to copy the repository.
+    repo : `str`
+        The location of the Gen 3 repository.
+    """
+    butler = daf_butler.Butler(repo)
+    with butler.export(directory=dataset.configLocation, format="yaml") as contents:
+        contents.saveDatasets(butler.registry.queryDatasets(datasetType=..., collections=..., expand=True))
+
+
+def _get_instrument_class(instrument):
+    """Convert a Gen 2 instrument name to a Gen 3 instrument class.
+
+    Parameters
+    ----------
+    instrument : `str`
+        A name in the format returned by
+        `lsst.obs.base.CameraMapper.getCameraName`.
+
+    Returns
+    -------
+    instrumentClass : `str`
+        The fully-qualified `~lsst.obs.base.Instrument` class for the
+        corresponding instrument.
+    """
+    classes = {
+        "decam": "lsst.obs.decam.DarkEnergyCamera",
+        "hsc": "lsst.obs.subaru.HyperSuprimeCam",
+    }
+
+    try:
+        return classes[instrument]
+    except KeyError:
+        raise ValueError(f"Unsupported instrument {instrument}; consider registering its Instrument class.")
```

Comment:
This might be a slightly misleading error message, if it's interpreted as a request to register a Gen3 `Instrument` class with the the registry (which of course won't modify this dict).

Reply:
Yes, it means adding it to the list here and not adding to registry. I've been worrying about short name to class name mapping for a long time. We've mitigated a bit now that registry can do that look up for you but we've never solved this problem in the general case where you are seeding a repo. If we are going to start adding dicts like this all over the code we should add them to the obs_base code that does this already by registry lookup (falling back to a dict if not registered). If we are doing that we could also remove the need for specifying an instrument class in 2to3 conversion by including a dict that maps gen2 mapper to instrument class.

Reply:
I'll reword the message to avoid using the word "register".

@timj, while I'd love to see a centralized/automated solution, note that in this case it's unlikely that we will have `ap_verify` datasets for instruments other than HSC, DECam, and LSSTCam (I'll deal with that case on DM-24844), at least not before we stop supporting Gen 2 in `ap_verify`. So a dict here is a bit more maintainable than it would be in general-purpose code, where it would need to be kept up to date with the full list of known obs packages.

I suggest leaving it as-is, and removing/re-iplementing once a centralized lookup is available.